music21/musicxml/m21ToXml.py

```diff
@@ -1404,7 +1404,7 @@ def __init__(self, score: Optional[stream.Score] = None, makeNotation: bool = Tr
 
         self.partExporterList: List['PartExporter'] = []
 
-        self.joinedGroups: List['StaffGroup'] = []
+        self.groupsToJoin: List['StaffGroup'] = []
```

Comment:
This attribute was introduced during v7 dev (cc09afcde9283538d74aef6bb5adf2fc7130c432) so that's why we are taking the opportunity to rename before final release to avoid implying the joining was already done at any given point.